src/main/java/com/amplitude/Response.java

```diff
@@ -0,0 +1,173 @@
+
+    private static Response createNewResponseFromJSON(JSONObject json) {
+        res = new Response();
+        int code  = json.getInt("code");
+        Status status = getCodeStatus(code);
+        res.code = code;
+        res.status = status;
+        if (status == Status.SUCCESS) {
+            res.SuccessBody = new JSONObject();
+            res.SuccessBody.put("eventsIngested", json.getInt("events_ingested"));
+            res.SuccessBody.put("payloadSizeBytes", json.getInt("payload_size_bytes"));
+            res.SuccessBody.put("serverUploadTime", json.getLong("server_upload_time"));
+        } else if (status == Status.INVALID) {
+            res.InvalidRequestBody = new JSONObject();
+            res.error = getStringValueWithKey(json, "error");
+            res.InvalidRequestBody.put("missingField", getStringValueWithKey(json, "missing_field"));
+            JSONObject eventsWithInvalidFields = getJSONObjectValueWithKey(json, "events_with_invalid_fields");
+            res.InvalidRequestBody.put("eventsWithInvalidFields", eventsWithInvalidFields);
+            JSONObject eventsWithMissingFields = getJSONObjectValueWithKey(json, "events_with_missing_fields");
+            res.InvalidRequestBody.put("eventsWithMissingFields", eventsWithMissingFields);
+        } else if (status == Status.PAYLOAD_TOO_LARGE) {
+            res.error = getStringValueWithKey(json, "error");
+        } else if (status == Status.RATELIMIT) {
+            res.error = getStringValueWithKey(json, "error");
+            res.RateLimitBody = new JSONObject();
+            res.RateLimitBody.put("epsThreshold", json.getInt("eps_threshold"));
+            JSONObject throttledDevices = getJSONObjectValueWithKey(json, "throttled_devices");
+            res.RateLimitBody.put("throttledDevices", throttledDevices);
+            JSONObject throttledUsers = getJSONObjectValueWithKey(json, "throttled_users");
+            res.RateLimitBody.put("throttledUsers", throttledUsers);
+            res.RateLimitBody.put("throttledEvents", convertJSONArrayToIntArray(json, "throttled_events"));
+            JSONObject exceededDailyQuotaDevices = getJSONObjectValueWithKey(json, "exceeded_daily_quota_devices");
+            res.RateLimitBody.put("exceededDailyQuotaDevices", exceededDailyQuotaDevices);
+            JSONObject exceededDailyQuotaUsers = getJSONObjectValueWithKey(json, "exceeded_daily_quota_users");
+            res.RateLimitBody.put("exceededDailyQuotaUsers", exceededDailyQuotaUsers);
+        } else if (status == Status.SYSTEM_ERROR) {
+            res.error = json.getString("error");
+            res.code = 0;
+        }
+        return res;
+    }
+
+    /*
+     * Use HTTPUrlConnection object to make async HTTP request,
+     * using data from event like device, class name, event props, etc.
+     *
+     * @return The response code
+     */
+    protected static Response syncHttpCallWithEventsBuffer(List<Event> events, String apiKey) {
```

Comment:
This function is to make a request. Why do we put it inside Response class?

Reply:
@dantetam  suggested that I should move the HTTP codes into here from Response.java. 
Should we create another class like HttpCall and move this in?

Reply:
Ideally Response class just handle the part like parsing, and make request calls could be in a separate class, and in future if we support more calls we can also add in that class.